### PR TITLE
Make ResourceTemplateForm a connected component

### PR DESCRIPTION
**Pull request by jcoyne**
This is needed so that we can load RDF from the server into the form

### CODE REVIEW

File: __tests__/components/editor/ResourceTemplateForm.test.js [hunk 1]
```diff
@@ -112,6 +112,7 @@ describe('<ResourceTemplateForm /> after fetching data from sinopia server', ()
   describe('configured component types', () => {
     it('renders a lookup component', async () => {
       const rtProps = {
+        rtId: 'resource:schema:Name',
```
Comment (ndushay) [inline __tests__/components/editor/ResourceTemplateForm.test.js]: If this is meant to be an actual resourceTemplateId, a better fake value would be 'myOrg:rt:myTemplate' - that is closer to what we will be seeing, and may make this less confusing to read in future.

File: __tests__/components/editor/ResourceTemplate.test.js [hunk 2]
```diff
@@ -1,61 +1,26 @@
...
-              valueDataType: {},
-              editable: 'true',
-              repeatable: 'false',
-              defaults: [],
-            },
-          },
-        ],
-      },
-    },
+  const resourceTemplate = {
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply __tests__/components/editor/ResourceTemplate.test.js]: It's a stub of how the resource in the redux state looks, so not a resource Template as returned from the server, but as we have transformed it in the redux state.

[the remark this answers is not in the document]
Comment (ndushay) [reply __tests__/components/editor/ResourceTemplate.test.js]: could the name of it somehow make that more clear?  mebbe `rtInRedux` or something?  Cause "resourceTemplate" means something specific to me when I don't see more context.  Or even a comment could clarify it for me.

[the remark this answers is not in the document]
Comment (jcoyne) [reply __tests__/components/editor/ResourceTemplate.test.js]: The only context that components should have is of the Redux state. Thus there should be no need to saying that it's in redux, as this should be clear.  Only reducers or actionCreators should have any awareness of trellis or any other part of the outside world.

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply __tests__/components/editor/ResourceTemplate.test.js]: @ndushay @jcoyne In this instance, the context that matters most to me (IMO!) is the "developer reading the code" context. In which case, I am in favor of renaming `resourceTemplate` for clarity. I don't care deeply about what the variable name is as long it conveys that the value represents an RT as stored in Redux.

[the remark this answers is not in the document]
Comment (jcoyne) [reply __tests__/components/editor/ResourceTemplate.test.js]: renamed to `reduxState`

File: __tests__/components/editor/ResourceTemplate.test.js [hunk 3]
```diff
@@ -1,61 +1,26 @@
...
-              valueDataType: {},
-              editable: 'true',
-              repeatable: 'false',
-              defaults: [],
-            },
-          },
-        ],
-      },
-    },
+  const reduxState = {
```
Comment (ndushay) [inline __tests__/components/editor/ResourceTemplate.test.js]: 👍  thank you.

**Pull request merged**